### PR TITLE
Refactor ClassLoader and ObjectManager singletons

### LINKED ISSUE #34: Refactor ClassLoader and ObjectManager singletons
Refactor the ClassLoader and ObjectManager and remove singletons. Instead store an instance of the class loader in the object manager and store an instance of the object manager in 'Kodekit'. Allow to set the debug and the cache through objects methods instead of setting them in the constructor.

Additionally add support for callStatic() to proxy static calls to the Kodekit class to the ObjectManager, eg Kodekit::getObject('com:foo.controller.bar');

### PR DESCRIPTION
**Pull request by johanjanssens**
resolve #34

### CODE REVIEW

**Comment by johanjanssens** [comment]
@ercanozkaya Can you review this ?

**Comment by ercanozkaya** [comment]
@johanjanssens This is looking good.

File: code/kodekit.php [hunk 1]
```diff
@@ -75,75 +82,81 @@ class Kodekit
...
+        /**
+         * Load the class locator
+         */
+        require_once dirname(__FILE__) . '/class/loader/loader.php';
 
-        //Setup the factory
-        $manager = Library\ObjectManager::getInstance($config);
+        $loader = new Library\ClassLoader();
+        $loader->setDebug(self::isDebug());
+        $loader->setCache(self::isCache());
```
Comment (oligriffiths) [inline code/kodekit.php]: Should it be possible to set the cache namespace here also?

Comment (johanjanssens) [inline code/kodekit.php]: Took that out for now. Need to review how cache namespace setting will work. Might get it directly from the env in a next step.

**Pull request merged**